api/v1alpha7/openstackmachine_webhook.go

```diff
@@ -62,6 +62,14 @@ func (r *OpenStackMachine) ValidateCreate() (admission.Warnings, error) {
 		allErrs = append(allErrs, field.Forbidden(field.NewPath("spec", "identityRef", "kind"), "must be a Secret"))
 	}
 
+	if r.Spec.RootVolume != nil && r.Spec.AdditionalBlockDevices != nil {
```

Comment:
this file has no unit tests, but I started to write it and i'll publish via another PR if reviewers don't mind.

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-openstack/issues/1737

Reply:
Ideally these would be envtests, btw. We could then re-use them to validate CEL.